### PR TITLE
Create Migrations with Relationships

### PR DESCRIPTION
**Pull request by mla-wtag**
[JTJ-44](https://welltravel.atlassian.net/browse/JTJ-44)

### CODE REVIEW

File: app/models/journal.rb [hunk 1]
```diff
@@ -1,2 +1,6 @@
 class Journal < ApplicationRecord
+
```
Comment (tgi-wtag) [inline app/models/journal.rb]: please cut out the superfluous spaces or at least follow a consistent pattern(always one line, always two lines).

Comment (tgi-wtag) [reply app/models/journal.rb]: this ofc applyes to all 14 files.

**Review by tgi-wtag — CHANGES_REQUESTED** [review]

**Review by tgi-wtag — APPROVED** [review]
looks ok to me

**Comment by meb-wtag** [comment]
All ok!

**Review by meb-wtag — APPROVED** [review]
all ok!

File: db/migrate/20230823111409_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+			t.timestamps
+		end
+
+	end
+
+	def down
+		drop_table :users
+	end
+	
+end
```
Comment (lra-wtag) [inline db/migrate/20230823111409_create_users.rb:19]: need formatting

File: db/migrate/20230823115839_create_journals.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class CreateJournals < ActiveRecord::Migration[7.0]
+
+	def up
```
Comment (lra-wtag) [inline db/migrate/20230823115839_create_journals.rb]: need formatting

File: db/migrate/20230823115839_create_journals.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+		end
+		
+		add_index :journals, :user_id
+	end
+
+	def down
+		drop_table :journals
+	end
+	
+end
```
Comment (lra-wtag) [inline db/migrate/20230823115839_create_journals.rb]: always add an extra line at the end of the file

File: db/migrate/20230823124431_create_tasks.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+class CreateTasks < ActiveRecord::Migration[7.0]
+
+	def up
```
Comment (lra-wtag) [inline db/migrate/20230823124431_create_tasks.rb]: need formatting

File: db/migrate/20230823111409_create_users.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+class CreateUsers < ActiveRecord::Migration[7.0]
+
+	def up 
+		create_table :users, id: false do |t|
```
Comment (lra-wtag) [inline db/migrate/20230823111409_create_users.rb]:
```suggestion
		create_table :users, do |t|
```

File: db/migrate/20230823111409_create_users.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+			t.integer :user_id, primary_key: true
```
Comment (lra-wtag) [inline db/migrate/20230823111409_create_users.rb]: in rails these ids get created automatically therefore we do not need these columns

File: db/migrate/20230823115839_create_journals.rb [hunk 8]
```diff
@@ -0,0 +1,22 @@
+class CreateJournals < ActiveRecord::Migration[7.0]
+
+	def up
+		create_table :journals, id: false do |t|
+			t.integer :journal_id, primary_key: true
```
Comment (lra-wtag) [inline db/migrate/20230823115839_create_journals.rb]: in rails these ids get created automatically therefore we do not need these columns

File: db/migrate/20230823115839_create_journals.rb [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+			t.integer :user_id
```
Comment (lra-wtag) [inline db/migrate/20230823115839_create_journals.rb]: is this the foreign key ? If so, then please define it as follows
```suggestion
			t.integer :user_id, foreign_key: true
```

File: db/migrate/20230823124431_create_tasks.rb [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+		create_table :tasks, id: false do |t|
+			t.integer :task_id, primary_key: true
+			t.integer :user_id
```
Comment (lra-wtag) [inline db/migrate/20230823124431_create_tasks.rb]: is this the foreign key ? If so, then please define it as follows
```suggestion
			t.integer :user_id, foreign_key: true
```

File: db/migrate/20230823124431_create_tasks.rb [hunk 11]
```diff
@@ -0,0 +1,21 @@
+class CreateTasks < ActiveRecord::Migration[7.0]
+
+	def up
+		create_table :tasks, id: false do |t|
+			t.integer :task_id, primary_key: true
```
Comment (lra-wtag) [inline db/migrate/20230823124431_create_tasks.rb]: in rails these ids get created automatically therefore we do not need these columns

File: db/migrate/20230823124431_create_tasks.rb [hunk 12]
```diff
@@ -0,0 +1,21 @@
+class CreateTasks < ActiveRecord::Migration[7.0]
+
+	def up
+		create_table :tasks, id: false do |t|
```
Comment (lra-wtag) [inline db/migrate/20230823124431_create_tasks.rb]:
```suggestion
		create_table :tasks do |t|
```

File: test/fixtures/journals.yml [hunk 13]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (lra-wtag) [inline test/fixtures/journals.yml]: no need of this file. please delete it

File: test/fixtures/tasks.yml [hunk 14]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (lra-wtag) [inline test/fixtures/tasks.yml]: no need of this file. please delete it

File: test/fixtures/users.yml [hunk 15]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (lra-wtag) [inline test/fixtures/users.yml]: no need of this file. please delete it

File: test/models/journal_test.rb [hunk 16]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (lra-wtag) [inline test/models/journal_test.rb]: no need of this file. please delete it

File: test/models/task_test.rb [hunk 17]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (lra-wtag) [inline test/models/task_test.rb]: no need of this file. please delete it

File: test/models/user_test.rb [hunk 18]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (lra-wtag) [inline test/models/user_test.rb]: no need of this file. please delete it

**Review by lra-wtag — CHANGES_REQUESTED** [review]

File: config/database.yml [hunk 19]
```diff
@@ -23,6 +23,7 @@ default: &default
 
 development:
   adapter : postgresql 
+  encoding: unicode
   database: journey_development
   username: michaellavelanet
```
Comment (meb-wtag) [inline config/database.yml]: Username and Password is not needed, or do you think so?

File: db/migrate/20230823111409_create_users.rb (hunk 6 above)
Comment (meb-wtag) [inline db/migrate/20230823111409_create_users.rb]: Formating rules have to be regarded, if not your collegues will be effected.

File: db/migrate/20230823111409_create_users.rb [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+			t.string :firstname, limit: 100, null: false
+			t.string :lastname, limit: 100, nul: false
```
Comment (tgi-wtag) [inline db/migrate/20230823111409_create_users.rb]: dont add limits here ( i dont know why, but laila told me so )

File: db/migrate/20230823115839_create_journals.rb [hunk 21]
```diff
@@ -0,0 +1,22 @@
+class CreateJournals < ActiveRecord::Migration[7.0]
+
+	def up
+		create_table :journals, id: false do |t|
+			t.integer :journal_id, primary_key: true
+			t.integer :user_id 
+			t.string :title
+			t.integer :date
+			t.text :content
+			t.text :goalstoday
```
Comment (tgi-wtag) [inline db/migrate/20230823115839_create_journals.rb]: t.text :goals_today
...

File: db/migrate/20230823124431_create_tasks.rb [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+			t.integer :task_id, primary_key: true
+			t.integer :user_id 
+			t.string :taskname
```
Comment (tgi-wtag) [inline db/migrate/20230823124431_create_tasks.rb]: task_name

**Review by meb-wtag — COMMENTED** [review]
Please work on formatting

File: db/migrate/20230823111409_create_users.rb [hunk 23]
```diff
@@ -0,0 +1,18 @@
+class CreateUsers < ActiveRecord::Migration[7.0]
+	def up 
+		create_table :users do |t|
+			t.string :firstname, null: false
+			t.string :lastname, nul: false
+			t.integer :employee_id
+			t.date :date_of_birth
+			t.date :joining_day
+			t.string :designation, limit: 50
```
Comment (lra-wtag) [inline db/migrate/20230823111409_create_users.rb]: no need of this validation in the db but in the model

File: db/migrate/20230823111409_create_users.rb [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+			t.binary :profile_picture, limit: 10.megabytes
```
Comment (lra-wtag) [inline db/migrate/20230823111409_create_users.rb]: no need of this validation in the db but in the model

File: db/migrate/20230823124431_create_tasks.rb [hunk 25]
```diff
@@ -0,0 +1,16 @@
+class CreateTasks < ActiveRecord::Migration[7.0]
+	def up
+		create_table :tasks do |t|
+			t.integer :user_id 
+			t.string :task_name
+			t.text :description
+	    	t.integer :date 
+	    	t.integer :status, limit: 3
```
Comment (lra-wtag) [inline db/migrate/20230823124431_create_tasks.rb]: no need of this validation in the db but in the model

**Review by lra-wtag — CHANGES_REQUESTED** [review]

File: app/controllers/journals_controller.rb [hunk 26]
```diff
@@ -0,0 +1,2 @@
+class JournalsController < ApplicationController
+end
```
Comment (lra-wtag) [inline app/controllers/journals_controller.rb:2]: we need the journal model as well as we have the journal controller and journal table

File: test/fixtures/journals.yml (hunk 13 above)
Comment (lra-wtag) [reply test/fixtures/journals.yml]: delete this file please

**Review by lra-wtag — CHANGES_REQUESTED** [review]

File: db/migrate/20230823111409_create_users.rb [hunk 27]
```diff
@@ -0,0 +1,18 @@
...
+			t.string :designation
+			t.binary :profile_picture
+			t.string :email
+			t.timestamps
+		end
+	end
+	def down
+		drop_table :users
+	end
+end
```
Comment (lra-wtag) [inline db/migrate/20230823111409_create_users.rb:19]: wrong indentations

File: db/migrate/20230823115839_create_journals.rb [hunk 28]
```diff
@@ -0,0 +1,16 @@
...
+			t.text :goals_today
+			t.text :goals_tomorrow
+			t.timestamps
+		end
+		add_index :journals, :user_id
+	end
+	def down
+		drop_table :journals
+	end	
+end
```
Comment (lra-wtag) [inline db/migrate/20230823115839_create_journals.rb]: wrong indentations

File: db/migrate/20230823124431_create_tasks.rb [hunk 29]
```diff
@@ -0,0 +1,15 @@
...
+	    	t.integer :date 
+			t.integer :status
+	    	t.timestamps
+		end
+		add_index :tasks , :user_id
+	end
+	def down
+		drop_table :tasks
+	end
+end
```
Comment (lra-wtag) [inline db/migrate/20230823124431_create_tasks.rb:15]: wrong indentations

File: db/schema.rb [hunk 30]
```diff
@@ -0,0 +1,53 @@
...
+    t.date "date_of_birth"
+    t.date "joining_day"
+    t.string "designation", limit: 50
+    t.binary "profile_picture"
+    t.string "email"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+end
```
Comment (lra-wtag) [inline db/schema.rb:49]: please run the migrations again

File: db/migrate/20230823115839_create_journals.rb [hunk 31]
```diff
@@ -0,0 +1,16 @@
...
+	def up
+		create_table :journals do |t| 
+			.string :title
+			t.integer :date
+			t.text :content
+			t.text :goals_today
+			t.text :goals_tomorrow
+			t.timestamps
+		end
+		add_index :journals, :user_id
```
Comment (lra-wtag) [inline db/migrate/20230823115839_create_journals.rb]: remove this

File: db/migrate/20230823124431_create_tasks.rb [hunk 32]
```diff
@@ -0,0 +1,15 @@
+class CreateTasks < ActiveRecord::Migration[7.0]
+	def up
+		create_table :tasks do |t|
+			t.string :task_name
+			t.text :description
+	    	t.integer :date 
+			t.integer :status
+	    	t.timestamps
+		end
+		add_index :tasks , :user_id
```
Comment (lra-wtag) [inline db/migrate/20230823124431_create_tasks.rb]: remove this

**Review by lra-wtag — CHANGES_REQUESTED** [review]

File: Gemfile [hunk 33]
```diff
@@ -10,7 +10,7 @@ gem "rails", "~> 7.0.7", ">= 7.0.7.2"
 gem "sprockets-rails"
 
 # Use postgresql as the database for Active Record
-gem "pg", "~> 1.1"
+gem "pg", "~> 1.2"
```
Comment (lra-wtag) [inline Gemfile]: no need to add any version
```suggestion
gem "pg"
```

**Review by lra-wtag — CHANGES_REQUESTED** [review]

**Review by lra-wtag — APPROVED** [review]

**Comment by mla-wtag** [comment]
Lettsss goooooooo

**Pull request merged**